src/server/scripts/Spells/spell_generic.cpp

```diff
@@ -1322,30 +1329,13 @@ class spell_gen_gift_of_naaru : public AuraScript
         if (!GetCaster())
             return;
 
-        float heal = 0.0f;
-        switch (GetSpellInfo()->SpellFamilyName)
+        if (SpellEffectInfo const* eff1 = GetSpellInfo()->GetEffect(EFFECT_1))
         {
-            case SPELLFAMILY_MAGE:
-            case SPELLFAMILY_WARLOCK:
-            case SPELLFAMILY_PRIEST:
-                heal = 1.885f * float(GetCaster()->SpellBaseDamageBonusDone(GetSpellInfo()->GetSchoolMask()));
-                break;
-            case SPELLFAMILY_PALADIN:
-            case SPELLFAMILY_SHAMAN:
-                heal = std::max(1.885f * float(GetCaster()->SpellBaseDamageBonusDone(GetSpellInfo()->GetSchoolMask())), 1.1f * float(GetCaster()->GetTotalAttackPowerValue(BASE_ATTACK)));
-                break;
-            case SPELLFAMILY_WARRIOR:
-            case SPELLFAMILY_HUNTER:
-            case SPELLFAMILY_DEATHKNIGHT:
-                heal = 1.1f * float(std::max(GetCaster()->GetTotalAttackPowerValue(BASE_ATTACK), GetCaster()->GetTotalAttackPowerValue(RANGED_ATTACK)));
-                break;
-            case SPELLFAMILY_GENERIC:
-            default:
-                break;
+            float healPct = eff1->CalcValue() / 100.0f;
+            float heal = 0.2f * GetCaster()->GetMaxHealth();
+            int32 healTick = std::floor(heal / aurEff->GetTotalTicks());
+            amount += int32(std::max(healTick, 0));
```

Comment:
we may also remove that std::max here i guess?

Reply:
why?

Reply:
healTick will always be >= 0

Reply:
Unless `eff->CalcValue()` is negative (changed in the future) ?

Reply:
that'd require us to fix the spell either way (again) in that scenario, either it'll heal exactly 0 or it will deal damage (if < 0) so we might actually drop std::max for some minimal performance improvement

Reply:
cool, will do